product_code/authentication_service/api/src/main/scala/de/upb/cs/uc4/authentication/api/AuthenticationService.scala

```diff
@@ -19,30 +18,53 @@ trait AuthenticationService extends UC4Service {
   override val name: String = "authentication"
   override val autoAcl: Boolean = false
 
-  /** Checks if the username and password pair exists */
-  def check(user: String, pw: String): ServiceCall[NotUsed, (String, AuthenticationRole)]
-
   /** Sets the authentication data of a user */
   def setAuthentication(): ServiceCall[AuthenticationUser, Done]
 
   /** Changes the password of the given user */
   def changePassword(username: String): ServiceCall[AuthenticationUser, Done]
 
+  /** Logins a user and return a refresh and a login token in the header */
+  def login: ServiceCall[NotUsed, Done]
+
+  /** Generates a new login token out of a refresh token */
+  def refresh: ServiceCall[NotUsed, JsonUsername]
+
+  /** Logs the user out */
+  def logout: ServiceCall[NotUsed, Done]
+
+  /** Allows GET */
+  def allowedGet: ServiceCall[NotUsed, Done]
+
   /** Allows PUT */
   def allowedPut: ServiceCall[NotUsed, Done]
 
   final override def descriptor: Descriptor = {
     import Service._
     super.descriptor
       .addCalls(
-        restCall(Method.GET, pathPrefix + "/users?user&pw", check _),
         restCall(Method.POST, pathPrefix + "/users", setAuthentication _),
+
         restCall(Method.PUT, pathPrefix + "/users/:username", changePassword _)(CustomMessageSerializer.jsValueFormatMessageSerializer, MessageSerializer.DoneMessageSerializer),
-        restCall(Method.OPTIONS, pathPrefix + "/users", allowedPut _)
+        restCall(Method.GET, pathPrefix + "/login", login _),
+        restCall(Method.GET, pathPrefix + "/refresh", refresh _),
+        restCall(Method.GET, pathPrefix + "/logout", logout _),
+
+        restCall(Method.OPTIONS, pathPrefix + "/users", allowedPut _),
```

Comment:
The parameter must also be specified in the corresponding OPTIONS call.

```suggestion
        restCall(Method.OPTIONS, pathPrefix + "/users/:username", allowedPut _),
```